lib/_emerge/actions.py

```diff
@@ -2790,15 +2792,26 @@ def relative_profile_path(portdir, abs_profile):
     return profilever
 
 
-def getportageversion(portdir, _unused, profile, chost, vardb):
-    pythonver = (
-        "python"
-        f" {sys.version_info[0]}"
-        f".{sys.version_info[1]}"
-        f".{sys.version_info[2]}"
-        f"-{sys.version_info[3]}"
-        f"-{sys.version_info[4]}"
-    )
+def get_libc_version(vardb: portage.dbapi.vartree.vardbapi) -> List[str]:
```

Comment:
I have no idea why this returns a list, to be honest. I don't think multiple providers of `virtual/libc` should ever be installed. Can you check git history for this to see why/when it was added and do a followup if appropriate? Thanks.

Reply:
Yeah, I had the same thought when committing it too. Then thought why change logic and continued with the flow. Will check.

Reply:
The implementation seems like multiple libc versions was considered and it was written originally by Zac.
```py
if libclist:
    for cpv in sorted(libclist):
        libc_split = portage.catpkgsplit(cpv)[1:]
        if libc_split[-1] == "r0":
            libc_split = libc_split[:-1]
        libcver.append("-".join(libc_split))
else:
    libcver = ["unavailable"]
return libcver
```
[This](https://stackoverflow.com/questions/847179/multiple-glibc-libraries-on-a-single-host) SO question also talks about multiple libc versions coexisting and `patchelf`, etc. Will someone downstream or someone cross compiling have multiple libc versions?  What should we do @thesamesam?